tests/infrastructure/src/main/java/org/mule/test/infrastructure/process/Controller.java

```diff
@@ -69,19 +72,30 @@ public Controller(String muleHome, int timeout) {
     this.domainsDir = new File(muleHome + "/domains");
     this.appsDir = new File(muleHome + "/apps/");
     this.libsDir = new File(muleHome + "/lib/user");
+    this.internalRepository = new File(muleHome, "repository");
     this.wrapperConf = Paths.get(muleHome + "/conf/wrapper.conf");
     this.timeout = timeout != 0 ? timeout : DEFAULT_TIMEOUT;
   }
 
   public abstract String getMuleBin();
 
   public void start(String[] args) {
+    checkRepositoryLocationAndUpdateInternalRepoPropertyIfPresent(args);
     int error = runSync("start", args);
     if (error != 0) {
       throw new MuleControllerException("The mule instance couldn't be started");
     }
   }
 
+  protected void checkRepositoryLocationAndUpdateInternalRepoPropertyIfPresent(String... args) {
+    Optional<String> repoVar =
+        Arrays.asList(args).stream().filter(arg -> arg.contains("-M-DmuleRuntimeConfig.maven.repositoryLocation=")).findFirst();
+    if (repoVar.isPresent()) {
+      this.internalRepository = new File(repoVar.get().split("=")[1]);
```

Comment:
I think that the internal is something to note that it is not the remote repositories. So maybe it has sense to keep it in the name. @gsfernandes What do you think??
Anyway a possible name without internal could be runtimeRepository to differentiate it from the app maven repository.